File: LivrusVirtus/wxWindows/wx.NET/Delphi.NET/fontes/TextValidatorCtrl.pas
unit TextValidatorCtrl;

interface

uses
  System.Drawing, wx;

type
  TTextValidatorCtrl = class(wx.TextCtrl)
  private
    nLocalFiltro: integer;
  public
    constructor Create(parent: wx.Window; id: integer; valor: string;
      pos: System.Drawing.Point; size: System.Drawing.Size;
      estilo, nfiltro: integer);
    procedure OnChar(sender: System.Object; e: wx.Event);
  end;

  [assembly: RuntimeRequiredAttribute(TypeOf(TTextValidatorCtrl))]

const
	FILTER_NONE: integer = 0;
	FILTER_ALPHA: integer = 1;
	FILTER_NUMERIC: integer = 2;

implementation

constructor TTextValidatorCtrl.Create(parent: wx.Window; id: integer;
  valor: string; pos: System.Drawing.Point; size: System.Drawing.Size;
  estilo, nfiltro: integer);
begin
  inherited Create(parent, id, valor, pos, size, estilo, nil, '');

  nLocalFiltro := nfiltro;
	EVT_CHAR( OnChar );
end;

procedure TTextValidatorCtrl.OnChar(sender: System.Object; e: wx.Event);
var ke: wx.KeyEvent;
  nChave: integer;
begin
  ke := wx.KeyEvent(e);
  nChave := ke.KeyCode;

  if (nLocalFiltro <> FILTER_NONE) then begin
    if (nChave < integer(wx.KeyCode.WXK_SPACE)) or (nChave = integer(wx.KeyCode.WXK_DELETE))
      or (nChave > 255) then
      e.Skip()
    else if (nLocalFiltro = FILTER_ALPHA) then begin
      if ((nChave >= 65) and (nChave <= 90)) or
         ((nChave >= 97) and (nChave <= 122))  then
        e.Skip();
    end else if (nLocalFiltro = FILTER_NUMERIC) then begin
      if (nChave >= 48) and (nChave <= 57) then
        e.Skip();
    end else
      e.Skip();
  end else
    e.Skip();
end;

end.

File: LivrusVirtus/wxWindows/wx.NET/VB.NET/TextValidatorCtrl.vb
Imports System
Imports System.Drawing
Imports wx

Namespace Livrus

	Public Class TextValidatorCtrl
		Inherits wx.TextCtrl
	
		public shared FILTER_NONE as integer = 0
		public shared FILTER_ALPHA as integer = 1
		public shared FILTER_NUMERIC as integer = 2
    private nLocalFiltro as integer

    Public Sub New(ByVal parent as Window, ByVal id as integer, _
      ByVal valor as string, ByVal pos as System.Drawing.Point, _
      ByVal size as System.Drawing.Size, ByVal estilo as long, _
      ByVal nfiltro as integer)
      MyBase.New(parent, id, valor, pos, size, estilo, nothing, "")

      nLocalFiltro = nfiltro
			EVT_CHAR( new EventListener(AddressOf OnChar ) )
    End Sub

		public sub OnChar(ByVal sender As Object, ByVal e As wx.Event)
      dim ke as wx.KeyEvent
      dim nChave as integer

      ke = CType(e, wx.KeyEvent)
      nChave = ke.KeyCode
    
      if (nLocalFiltro <> FILTER_NONE) then 
        if (nChave < wx.KeyCode.WXK_SPACE) or _
          (nChave = wx.KeyCode.WXK_DELETE) or (nChave > 255) then
          e.Skip()
        else if (nLocalFiltro = FILTER_ALPHA) then 
          if (nChave >= 65 and nChave <= 90) or (nChave >= 97 and nChave <= 122)  then
            e.Skip()
          end if
        else if (nLocalFiltro = FILTER_NUMERIC) then
          if (nChave >= 48 and nChave <= 57) then
            e.Skip()
          end if
        else
          e.Skip()
        end if
      else
        e.Skip()
      end if
		End Sub

	End Class

End Namespace
